src/Storage/Services/DataService.cs

```diff
@@ -71,12 +71,12 @@ public async Task<(string FileHash, ServiceError ServiceError)> GenerateSha256Ha
         }
 
         /// <inheritdoc/>
-        public async Task UploadDataAndCreateDataElement(string org, Stream stream, DataElement dataElement)
+        public async Task UploadDataAndCreateDataElement(string org, Stream stream, DataElement dataElement, long internalId)
```

Comment:
unclear which objects internalId we are reffering to here.

Reply:
Have renamed to instanceInternalId now. The preferred solution would be to redesign the interfaces so that the id would be internal to the repository methods, but I wouldn’t like to introduce such a change in the first release.